### PR TITLE
Migrate to DSL2 and start modularizing

### PR DESCRIPTION
**Pull request by fanninpm**
This adds more lines to the workflow, but it separates the process definitions from the logic that shows how the individual processes fit together (in the `workflow` block on the bottom).

EDIT: All the process declarations are now in separate files.

### CODE REVIEW

File: main.nf [hunk 1]
```diff
@@ -22,1058 +25,209 @@ params.container_pangolin = 'staphb/pangolin:latest'
...
+    REMOVE_JUNK_DELS(BBMAP_ALIGN.out.aligned_sams, GRAB_NEXTCLADE_DATA.out.reference_genome) // out: filtered_junk_sams, _
+    REMOVE_SINGLETONS(REMOVE_JUNK_DELS.out.filtered_junk_sams, GRAB_NEXTCLADE_DATA.out.reference_genome) // out: filtered_singletons_sams, _
+    SOFT_TRIM(REMOVE_SINGLETONS.out.filtered_singletons_sams) // out: soft_trimmed_sams, _
+    PILEUP(SOFT_TRIM.out.soft_trimmed_sams) // out: basecov, covstats, hist, _
+    SUMMARIZE_COVERAGE(PILEUP.out.basecov.toSortedList()) // out: coverage_summary, _
+
+    // SAMTOOLS AND IVAR
+    // =================
+    SAMTOOLS_SORT_INDEX(SOFT_TRIM.out.soft_trimmed_sams) // out: bamfile, bamfile_index, _
+    IVAR_CONSENSUS(SAMTOOLS_SORT_INDEX.out.bamfile) // out: consensus, consensus_collect, _
```
Comment (fanninpm) [inline main.nf:192]: TODO: bring back overall consensus FASTA file. Something like:

```suggestion
    IVAR_CONSENSUS(SAMTOOLS_SORT_INDEX.out.bamfile) // out: consensus, consensus_collect, _
    IVAR_CONSENSUS.out.consensus_collect |
        collectFile(
            name: "${file(params.outdir).getSimpleName()}_consensus.fasta",
            sort: true,
            storeDir: "${params.outdir}"
        )
```

Comment (fanninpm) [reply main.nf:192]: Resolved in ffd88f0

**Comment by friesac** [comment]
Great work @fanninpm. This all looks pretty straightforward and less complicated than I had envisioned DSL 2 syntax to be. I have one suggestion now that the modules are in place, I would annotate your workflow as much as humanly possible to help others as we attempt to have them understand the process structure in the workflows with the eventual goal of having someone help you replicate this structure for a different pathogen.

**Comment by friesac** [comment]
I'll try and go through the review tomorrow as time permits.

**Pull request merged**